### PR TITLE
Incorrect username handling

### PR DESCRIPTION
**Pull request by JamesADoggett**
Loops until valid username entered

### CODE REVIEW

**Review by acairns-345 — APPROVED** [review]
All approved, thanks for your work on this! 

As we discussed, please wait for Lukasz to merge the checkPassword branch into yours before merging the incorrectUsernameHandling branch :)

**Comment by Wamacava** [comment]
I just merged my branch with yours :)

**Review by Wamacava — APPROVED** [review]

**Pull request merged**
